android/java/org/chromium/chrome/browser/ntp/sponsored/NTPUtil.java

```diff
@@ -0,0 +1,269 @@
+                        rewardsBottomSheetDialogFragment.show(chromeActivity.getSupportFragmentManager(), "rewards_bottom_sheet_dialog_fragment");
+                        rewardsBottomSheetDialogFragment.setCancelable(false);
+
+                        sponsoredTab.updateBannerPref();
+                    }
+                });
+            
+                turnOnAdsButton.setOnClickListener(new View.OnClickListener() {
+                    @Override
+                    public void onClick(View view) {
+                        NTPUtil.turnOnAds();
+                        nonDistruptiveBannerLayout.setVisibility(View.GONE);
+
+                        sponsoredTab.updateBannerPref();
+                    }
+                });
+
+                switch(ntpType) {
+                    case SponsoredImageUtil.BR_OFF:
+                        bannerText.setText(chromeActivity.getResources().getString(R.string.get_paid_to_see_image));
+                        learnMoreText.setVisibility(View.VISIBLE);
+                        break;
+                    case SponsoredImageUtil.BR_ON_ADS_OFF:
+                        bannerText.setText(chromeActivity.getResources().getString(R.string.get_paid_to_see_image));
+                        learnMoreText.setVisibility(View.VISIBLE);
+                        break;
+                    case SponsoredImageUtil.BR_ON_ADS_OFF_BG_IMAGE:
+                        bannerText.setText(chromeActivity.getResources().getString(R.string.you_can_support_creators));
+                        turnOnAdsButton.setVisibility(View.VISIBLE);
+                        break;
+                    case SponsoredImageUtil.BR_ON_ADS_ON:
+                        bannerText.setText(chromeActivity.getResources().getString(R.string.you_are_getting_paid));
+                        learnMoreText.setVisibility(View.VISIBLE);
+                        break;
+                }
+            }
+        }, 1500);
+    }
+
+    public static NTPImage getNTPImage() {
+        SharedPreferences mSharedPreferences = ContextUtils.getAppSharedPreferences();
+
+        if (mSharedPreferences.getBoolean(BackgroundImagesPreferences.PREF_SHOW_SPONSORED_IMAGES, true)
+            && SponsoredImageUtil.getSponsoredImages().size() > 0
+            && mSharedPreferences.getInt(BackgroundImagesPreferences.PREF_APP_OPEN_COUNT, 0) == 1
+            && SponsoredImageUtil.getTabIndex() == 2
+            && BraveAdsNativeHelper.nativeIsLocaleValid(Profile.getLastUsedProfile())
+            && !BravePrefServiceBridge.getInstance().getSafetynetCheckFailed()
+            && ChromeFeatureList.isEnabled(BraveFeatureList.BRAVE_REWARDS)) {
+            SponsoredImage sponsoredImage = SponsoredImageUtil.getSponsoredImage();
+            SponsoredImageUtil.incrementTabIndex(3);
+            return sponsoredImage;
+        }
+
+        if (mSharedPreferences.getBoolean(BackgroundImagesPreferences.PREF_SHOW_SPONSORED_IMAGES, true)
```

Comment:
there are way too many duplicated conditions here, the common conditions should be combined and then use a nested if